### PR TITLE
Fix aggregation handling and example

### PR DESCRIPTION
**Pull request by ymotongpoo**
Fixed error on submitting Observer value to the backend and fixed the example to eliminate the unrequired variable.

### CODE REVIEW

File: exporter/metric/metric.go [hunk 1]
```diff
@@ -374,12 +370,30 @@ func recordToTypedValueAndTimestamp(r *export.Record) (*monitoringpb.TypedValue,
...
+				Seconds: now,
 			},
-		}, t, nil
+		}
+		switch kind {
+		case apimetric.Int64NumberKind:
+			return &monitoringpb.TypedValue{
+				Value: &monitoringpb.TypedValue_Int64Value{
+					Int64Value: value,
+				},
```
Comment (rghetia) [inline exporter/metric/metric.go:382]: This code is repeated so may be it could be converted to a func.

**Comment by rghetia** [comment]
cc/ @nilebox

File: exporter/metric/metric.go [hunk 2]
```diff
@@ -374,12 +370,30 @@ func recordToTypedValueAndTimestamp(r *export.Record) (*monitoringpb.TypedValue,
 			StartTime: &googlepb.Timestamp{
 				Seconds: now,
 			},
-		}
-		return &monitoringpb.TypedValue{
-			Value: &monitoringpb.TypedValue_Int64Value{
-				Int64Value: value,
+			EndTime: &googlepb.Timestamp{
+				Seconds: now,
```
Comment (nilebox) [inline exporter/metric/metric.go:374]: This is to be later fixed in #25 as far as I understand.

File: exporter/metric/metric.go [hunk 3]
```diff
@@ -374,12 +370,30 @@ func recordToTypedValueAndTimestamp(r *export.Record) (*monitoringpb.TypedValue,
...
 			},
-		}, t, nil
+		}
+		switch kind {
```
Comment (nilebox) [inline exporter/metric/metric.go:377]: You have a repetitive code `return ..., t, nil` below.
If you extract this switch-case block into a separate function `countToTypedValue(kind, value)`, the code will be much cleaner:
```go
typedValue := countToTypedValue(kind, value)
return typedValue, t, nil
```

Comment (nilebox) [reply exporter/metric/metric.go:377]: You can also do the same for LastValue and Sum below.
Actually would also be nice to extract every if condition there:
```
if count, ok := agg.(aggregator.Count); ok {
  return countToTypedValueAndTimestamp(count, kind)
} else if lv, ok := agg.(aggregator.LastValue); ok {
  return lastValueToTypedValueAndTimestamp(lv, kind)
} else ...
```

File: exporter/metric/metric.go (hunk 1 above)
Comment (ymotongpoo) [reply exporter/metric/metric.go:382]: Worrysome part here is that only Counter returns int64 where as the other kinds return apimetric.Number.

The combination of this and the comment by @nilebox looks better, i.e. extracting each cases to functions and in addition to it make TypeValue part a function.

WDYT?

File: exporter/metric/metric.go (hunk 3 above)
Comment (ymotongpoo) [reply exporter/metric/metric.go]: Thank you, Nail. I extracted all cases to functions as suggested. Actually it looks much cleaner.

**Pull request merged**